### DISCUSSION TITLE
Discussion about GSoC projects

### DISCUSSION THREAD

**Discussion by Mu-Magdy** [weecology/retriever#1682] [General]
Hi There,

I'm really keen on your GSoC projects after following your work last year. I'm having trouble choosing one because they're all so interesting, and I believe I have the skills for many of them. Is there a project you're eager to prioritize? Knowing this would help me decide.

**Comment by henrykironde** [comment]
We don't prioritize projects; our main goal is to find individuals capable of delivering and completing them. Therefore, we typically select at least one contributor per project, with preference given to those consistently delivering substantial contributions. While minor contributions such as fixing typos are appreciated, they usually don't weigh heavily in the selection process. Additionally, contributions don't necessarily have to be merged to be considered substantial.

**Reply by Mu-Magdy** [reply]
Thanks @henrykironde your reply motivated me to choose a project idea I like,
 I think the  (Advancing Bird Detection and Classification in Hand-Held Airborne Imagery) is a project i would like to work on.
 But can you give me an overview on what the current model is and do you have something in mind to work on or I  will be able to explore new ways.

 
 And about the proposal. do I need to send it to you first or i submit it to GSoC once finished?

**Reply by henrykironde** [reply]
> And about the proposal. do I need to send it to you first or i submit it to GSoC once finished? 

You can draft your proposal within the Google dashboard, allowing me to provide feedback and suggestions prior to your submission. Let me know once the draft is up.

> But can you give me an overview on what the current model is and do you have something in mind to work on or I will be able to explore new ways.
 
I will let @bw4sz or @ethanwhite drop a few lines, else I will get back to you tomorrow.

**Reply by Mu-Magdy** [reply]
Thanks @henrykironde.

I don't know what is google dashboard I think you mean google docs. So Here is the [link](https://docs.google.com/document/d/1cJh5Cb3cJxQIZoXb1NF-ejZAnO_cJ6CRTZZcNjeMhEc/edit?usp=sharing) for the draft. I would really appreciate your feedback

**Reply by Mu-Magdy** [reply]
I can modify it if you want to or I can restructure it completely or extend the timeline to add more weeks if needed, But I am afraid to miss the deadline for the proposal submission.

**Reply by Mu-Magdy** [reply]
Hi @henrykironde,

Is there a modification in the proposal? 
Should I submit it or not?

**Reply by ethanwhite** [reply]
Hi @Mu-Magdy - I have it a quick read and everything looks good! Definitely submit before the deadline even if you don't hear from Henry. Good luck!

**Reply by henrykironde** [reply]
Looks good to me. Sent you an email

**Reply by Mu-Magdy** [reply]
Hi @henrykironde, 
What do you advice me to do now?
I read the papers @bw4sz mentioned in his comment. And explored the model that he shared.
How to better prepare for this project ?

**Comment by bw4sz** [comment]
Hello @Mu-Magdy here are some details on the status of the project.

1. We have a large number (1000s) of photos taken from airplanes of wading bird colonies. Here is a low resolution screenshot

![Screenshot 2024-03-28 at 12 59 59 PM](https://github.com/weecology/retriever/assets/1208492/2a9e6a25-0d29-4840-81b8-9c6b98781603)

2. We have images and a trained model from UAV data. These images are higher resolution and easier to identify. I can share the trained model with you. This UAV model works reasonably well on Airplane imagery. Sample predicted airplane image ->

![Screenshot 2024-03-28 at 1 03 51 PM](https://github.com/weecology/retriever/assets/1208492/91e3e1d6-0bb0-4f4f-b747-e9869e40f771)

4. We have annotations from scientists on Airplane imagery that you can use for training and testing. We can finetune the UAV model for lower resolution airplane data. This is the first step.

Depending on your interest, there are a number of avenues from there. Here is one option depending on time investment and interesting. 

* There are many images of the same colony as the photographer flies around it. If we have 30 images of the same colony, can we estimate the total number of unique birds in the colony? Some interesting papers (https://www.mdpi.com/2073-4395/12/2/440, https://arxiv.org/pdf/2012.00946.pdf, https://www.ecva.net/papers/eccv_2022/papers_ECCV/papers/136690224.pdf)

This would be important for the field of ecological monitoring at large. There are many times we have unstructured photos from a single 'unit', that we desire the total count. There are many ways to go about it, from re-identification, to temporal overlay, to image-matching, etc. A fun place to explore. I have other options as well that we can discuss.

**Reply by Mu-Magdy** [reply]
Thanks alot @bw4sz,
That was very helpful, 

But to make sure I understand correctly, Is the images annotated or I need to make time in the proposal to annotate data?  As this `'An annotated dataset for training and testing, contributing to the improvement of the model's performance.'` was written in the original project idea file as an expected output

And the part related to bird counting, Is the expected output a model to count the birds in the 'unit' or a comparison to different approaches or papers to choose the best on a later stage? or both is expected

**Reply by Mu-Magdy** [reply]
Hi @bw4sz,

Hope you are fine, 
Can you please share the trained model and any other relevant work to prepare for next steps?

**Comment by bw4sz** [comment]
The trained model is here: https://www.dropbox.com/scl/fi/e19bll7t1e8aigirj8vtx/species_model.pl?rlkey=culy6hqgma3esc06h49cty6r3&dl=0

It can be opened like so:


```
from deepforest import main
m = main.deepforest.load_from_checkpoint(model_path)
```
